apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx

```diff
@@ -112,51 +126,60 @@ const PostgrestConfig = () => {
+                              onChange={(event) => {
+                                let updatedValues: any = values
+                                updatedValues.db_schema = event.join(', ')
+                                resetForm({ values: updatedValues, initialValues: updatedValues })
+                                updateConfig({ ...updatedValues })
+                              }}
+                            />
+                            <p className="text-foreground-lighter text-sm">
                               The schemas to expose in your API. Tables, views and stored procedures
                               in these schemas will get API endpoints.
-                              <code className="text-xs">public</code> and{' '}
-                              <code className="text-xs">storage</code> are protected by default.
-                            </>
-                          }
-                          emptyMessage={
-                            <>
-                              <IconAlertCircle strokeWidth={2} />
-                              <div className="flex flex-col mt-2 text-center">
-                                <p className="text-sm align-center">
-                                  No schema available to choose
-                                </p>
-                                <p className="text-xs opacity-50">
-                                  New schemas you create will appear here
-                                </p>
-                              </div>
-                            </>
-                          }
-                          // value must be passed as array of strings
-                          value={(values?.db_schema ?? '').replace(/ /g, '').split(',')}
-                          // onChange returns array of strings
-                          onChange={(event) => {
-                            let updatedValues: any = values
-                            updatedValues.db_schema = event.join(', ')
-                            resetForm({ values: updatedValues, initialValues: updatedValues })
-                            updateConfig({ ...updatedValues })
-                          }}
-                        />
-                      )}
-                    </FormSectionContent>
+                            </p>
+
+                            {!isPublicSchemaEnabled && (
+                              <Alert_Shadcn_ variant="default">
+                                <Info className="h-4 w-4" />
+                                <AlertTitle_Shadcn_>
+                                  <ReactMarkdown>The `public` schema is not exposed</ReactMarkdown>
+                                </AlertTitle_Shadcn_>
+                                <AlertDescription_Shadcn_ className="flex flex-col gap-3">
+                                  <ReactMarkdown>
+                                    API endpoints will not be created for the `public` schema. You
```

Comment:
You will not be able to query tables and views in the public schema via postgREST and supabase-js.